connector_importer/components/importer_csv_std.py

```diff
@@ -0,0 +1,165 @@
+
+        data = [
+            [line[fieldname] for fieldname in fieldnames]
+            for line in lines
+        ]
+        return fieldnames, data
+
+    def run(self, record, is_last_importer=True, **kw):
+        """Run record job.
+
+        Steps:
+
+        * for each record, check if it is already imported or not and reference
+          them as created or updated
+        * launch the import with 'load()' method
+        * analyse error messages returned by 'load()' and remove relevant
+          references from the first step + create log error for them
+        * produce a report and store it on recordset
+        """
+
+        self.record = record
+        if not self.record:
+            # maybe deleted???
+            msg = 'NO RECORD FOUND, maybe deleted? Check your jobs!'
+            logger.error(msg)
+            return
+
+        self._init_importer(self.record.recordset_id)
+
+        mapped_lines = []
+        tracker_data = {
+            'created': {
+                # line_nr: (values, line, odoo_record),
+            },
+            'updated': {
+                # line_nr: (values, line, odoo_record),
+            },
+        }
+        lines = self._record_lines()
+        for i, line in enumerate(lines):
+            line = self.prepare_line(line)
+            options = self._load_mapper_options()
+            try:
+                with self.env.cr.savepoint():
+                    values = self.mapper.map_record(line).values(**options)
+                logger.debug(values)
+            except Exception as err:
+                values = {}
+                self.tracker.log_error(
+                    values, line, odoo_record=None, message=err)
+                if self._break_on_error:
+                    raise
+                continue
+            # Collect tracker data for later
+            # We store the parameters for chunk_report.track_{created,updated}
+            # functions, excepted the odoo_record which could not be known
+            # for newly created records
+            odoo_record_exists = self.record_handler.odoo_exists(
+                values, line, use_xmlid=True)
+            if odoo_record_exists:
+                odoo_record = self.record_handler.odoo_find(
+                    values, line, use_xmlid=True)
+                tracker_data['updated'][i] = [line, odoo_record]
+            else:
+                tracker_data['created'][i] = [line, None]
+
+            # handle forced skipping
+            skip_info = self.skip_it(values, line)
+            if skip_info:
+                self.tracker.log_skipped(values, line, skip_info)
+                continue
+            mapped_lines.append(values)
+
+        if mapped_lines:
+            try:
+                with self.env.cr.savepoint():
+                    fieldnames, data = self.prepare_load_params(mapped_lines)
+                    load_res = self.model.load(fieldnames, data)
+
+                    # Log load errors
```

Comment:
Extend this comment:
```py
# In case of errors `load` returns a list of messages with the cause
# and the rows range. Here we map these messages to tracked data
# and update the references to be able to provide a precise report.
```